core/src/test/scala/integration/kafka/api/SaslClientsWithInvalidCredentialsTest.scala

```diff
@@ -74,48 +75,58 @@ class SaslClientsWithInvalidCredentialsTest extends IntegrationTestHarness with
   @Test
   def testProducerWithAuthenticationFailure(): Unit = {
     val producer = createProducer()
-    verifyAuthenticationException(sendOneRecord(producer, maxWaitMs = 10000))
-    verifyAuthenticationException(producer.partitionsFor(topic))
+    try {
+      verifyAuthenticationException(sendOneRecord(producer, maxWaitMs = 10000))
+      verifyAuthenticationException(producer.partitionsFor(topic))
 
-    createClientCredential()
-    verifyWithRetry(sendOneRecord(producer))
+      createClientCredential()
+      verifyWithRetry(sendOneRecord(producer))
+    } finally producer.close()
```

Comment:
@akatona84 Thanks for the PR.  Normally producer/consumer instances created in the test are closed in [IntegrationTestHarness.tearDown ](https://github.com/confluentinc/ce-kafka/blob/master/core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala#L152). many core tests extends `IntegrationTestHarness` class.
Do we know, which tests were failed?

Reply:
Oh, right, it was suspicious that I did not see consumer thread on the failed .classMethod cases :) Thanks.
 testConsumerGroupServiceWithAuthenticationSuccess failed and left adminclient threads for later ones:
```
java.lang.AssertionError: Found unexpected threads during @AfterClass, allThreads=Set(Test worker, metrics-meter-tick-thread-2, main, Signal Dispatcher, Reference Handler, Finalizer, /0:0:0:0:0:0:0:1:55620 to /0:0:0:0:0:0:0:1:33045 workers Thread 2, kafka-admin-client-thread | adminclient-45, /0:0:0:0:0:0:0:1:55620 to /0:0:0:0:0:0:0:1:33045 workers Thread 3, metrics-meter-tick-thread-1), unexpected=Set(kafka-admin-client-thread | adminclient-45)
```